src/components/buttons/secondary-icon-button/secondary-icon-button.js

```diff
@@ -1,18 +1,23 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import filterAriaAttributes from '../../../utils/filter-aria-attributes';
-import filterDataAttributes from '../../../utils/filter-data-attributes';
+import omit from 'lodash/omit';
+import filterInvalidAttributes from '../../../utils/filter-invalid-attributes';
 import AccessibleButton from '../accessible-button';
 import { getBaseStyles } from './secondary-icon-button.styles';
 
+const propsToOmit = ['type'];
+
 export const SecondaryIconButton = props => {
   const buttonAttributes = {
+    ...filterInvalidAttributes(omit(props, propsToOmit)),
     'data-track-component': 'SecondaryIconButton',
-    ...filterAriaAttributes(props),
-    ...filterDataAttributes(props),
+    // if there is a divergence between `isDisabled` and `disabled`,
+    // we fall back to `isDisabled`
+    disabled: props.isDisabled,
```

Comment:
@jonnybel is this what you had in mind?

Reply:
It's one way to solve it, yeah 😄 Thanks!